msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/PublicClientApplication.java

```diff
@@ -112,17 +122,48 @@ public CompletableFuture<IAuthenticationResult> acquireToken(InteractiveRequestP
                 this,
                 context);
 
-        CompletableFuture<IAuthenticationResult> future = executeRequest(interactiveRequest);
-        futureReference.set(future);
+        CompletableFuture<IAuthenticationResult> future = new CompletableFuture<>();
+
+        if (this.broker != null) {
+            broker.acquireToken(this, parameters, future);
+            futureReference.set(future);
+        } else {
+            future = executeRequest(interactiveRequest);
+            futureReference.set(future);
```

Comment:
Since this is common between both conditions, it can be moved out of the if-else block.

Reply:
Fixed in the latest commit.